docs/docs/Best_Practices/Message_Limits.md

```diff
@@ -0,0 +1,27 @@
+# Message Limits
+
+There are limits on how much data can be sent back and forth between the plugin and engine at a time. There are five scenarios where this comes into play:
+1. Inputs sent from plugin to engine, as arguments to a [Plugin Operation](/References/Plugin_Operations.md). For example, the schema-defined `Repository` object that is provided as input to plugin operations.
+2. Outputs sent back from the plugin to the engine, as the return values from plugin operations.
+3. Exception messages and call stacks thrown by plugin code. For example, the `message` field within [User Visible Errors](/Best_Practices/User_Visible_Errors.md).
+4. Inputs sent from plugin to engine, as arguments to a [Platform library](/References/Platform_Libraries.md) function. For example, the `message` field that is passed to `logger.debug`. 
+5. Outputs sent back from the plugin to the engine, as the return values from Platform Library functions. For example, the `stdout` resulting from a call to `libs.run_bash`. 
+
+For case 1 and 2, the total size of data must be less than 4 mebibytes (4 MiB).
+
+For case 3, the total size of data must be less than 128 kibibytes (128 KiB).
```

Comment:
just curious, why is this lower?

Reply:
Yeah, case 1 and 2 are lower because we are still using the default message size when creating the Netty channel for DockerPluginRuntime.

Found the default within the Netty channel builder code:
```
The default maximum uncompressed size (in bytes) for inbound messages. Defaults to 4 MiB.
```